addons/mail/models/mail_thread.py

```diff
@@ -1199,7 +1199,10 @@ def _filter_excluded_local_part(email):
                 for alias in dest_aliases:
                     user_id = self._mail_find_user_for_gateway(email_from, alias=alias).id or self._uid
                     route = (alias.sudo().alias_model_id.model, alias.alias_force_thread_id, ast.literal_eval(alias.alias_defaults), user_id, alias)
-                    route = self._routing_check_route(message, message_dict, route, raise_exception=True)
+                    if hasattr(self.env[route[0]], '_routing_check_route'):
+                        route = self.env[route[0]]._routing_check_route(message, message_dict, route, raise_exception=True)
+                    else:
+                        route = self._routing_check_route(message, message_dict, route, raise_exception=True)
```

Comment:
Actually we could probably try to always use AliasModel if it exists and has the method (aka inherit from mail.thread), so that it enables overrides for all alias models, something like
```suggestion
                    AliasModel = self.env[route[0]] if route[0] in self.env and '_routing_check_route' in self.env[route[0]] else self
                    route = AliasModel._routing_check_route(message, message_dict, route, raise_exception=True)
```

Reply:
mhhh I think my implementation enable overrides for all alias models :thinking: 

The problem with that
```py
AliasModel = self.env[route[0]] if route[0] in self.env and '_routing_check_route' in self.env[route[0]] else self`
```
This part `'_routing_check_route' in self.env[route[0]]` will always be false. But I can do `callable(self.env[route[0]]._routing_check_route)` instead.

I will do it :+1:

Reply:
Actually I meant "hasattr" :) indeed "in" is for fields

Reply:
Yes, done :)